### ISSUE TITLE
Elasticsearch 2.0 configuration publish host doesn't work

### ISSUE THREAD

**Issue by gujason721** [elastic/elasticsearch#14408]
Hi,
In my configuration, I added following entries in elasticsearch.yml based on https://www.elastic.co/guide/en/elasticsearch/reference/2.0/modules-http.html

http.publish_host: xx.xx.xx.xx

http.publish_port: 9200

However, it doesn't work as in ES 1.7.

Only setting network.host: xx.xx.xx.xx allows for remote http connection.

**Comment by harpreetsb** [comment]
yes same problem, cat access it via public ip

**Comment by nik9000** [comment]
Sorry, I don't understand what you mean. Could you rephrase it?

**Comment by rmuir** [comment]
I think the confusion stems from `publish_XXX` not being listed in the documentation separately as an "advanced option". Setting the publish host does not change binding at all, it only changes what the node "advertises" and is only intended for special proxy/firewall situations. In 99% of cases people should not be changing it.

**Comment by nik9000** [comment]
> In 99% of cases people should not be changing it.

To make this as clear as possible: in almost all cases what you should change is the `host` setting. Not `bind_host` or `publish_host`.

Honestly https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-network.html could do with a rewrite to mention `host` first and way lower down mention that you can, in extraordinary situations, bind to a different host than you publish.

**Comment by rmuir** [comment]
+1

**Comment by avisri** [comment]
Config:
#rpm -qa  | grep elastic
elasticsearch-2.0.0-1.noarch
#grep -v ^# /etc/elasticsearch/elasticsearch.yml| sed  /^$/d| grep publish_host | sed -s "s/10.*/10.x.x.x/"
network.publish_host: 10.x.x.x

Result:
#netstat -planet | grep 9300   | less
tcp        0      0 ::ffff:127.0.0.1:9300       :::*                        LISTEN      495        30774459   26642/java

I was wondering whether it should bind to  10.x.x.x address instead ?

**Comment by nik9000** [comment]
> network.publish_host: 10.x.x.x

Use `network.host` instead.

**Comment by avisri** [comment]
Need to ensure 
Elastic listen locally only (9200) but talk to other nodes on the named interface ( 9300)

**Comment by clintongormley** [comment]
Closed by https://github.com/elastic/elasticsearch/pull/15360

New docs can be seen here: https://www.elastic.co/guide/en/elasticsearch/reference/2.x/modules-network.html
